docs/DeveloperGuide.md

```diff
@@ -1,25 +1,290 @@
 # Developer Guide
 
-## Design & implementation
+## Table of content
+**1. Setting Up**\
+**2. Design**\
+**3. Implementation**
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## 1. Setting Up
 
+**Prerequisites:**
+* JDK 11
+* IntelliJ IDE
+
+Refer to the guide [_Setting Up_](SettingUp.md).
+
+## 2. Design
+
+### 2.1 Architecture
+The **Architecture Diagram** below shows a high-level design of WhereGotTime. The design is based 
+on multi-tier architecture where the processing, UI, logic and data management are separated. The upper
+tiers make use of services provided by lower tiers.
+
+![](images/Overall Architecture.png)
+
+<div markdown="span" class="alert alert-primary">
+
+**Tip:** The `.puml` files used to create diagrams in this document can be found in the [diagrams](https://github.com/se-edu/addressbook-level3/tree/master/docs/diagrams/) folder. Refer to the [_PlantUML Tutorial_](https://se-education.org/guides/tutorials/plantUml.html) to learn to create and edit diagrams.
+
+</div>
+
+Here is a quick overview of the main components of WhereGotTime:
+1. `WhereGotTime` - initializes the components in the app
+1. `UI` - the user interface of the app
+1. `Parser` - analyzes and interprets user input
+1. `Command` - executes the action required by user
+1. `Storage` - reads and writes data to file 
+1. `Common` - collection of classes used by multiple components
+   * `Users` - the users who have logged into WhereGotTime
+   * `Timetable` - the timetable of each of the users
+   * `Event` - an object representing a lesson in the timetable
+
+**How architecture components interact**
+
+The following Sequence Diagram displays how components interact when the user inputs `clear /fri`
+
+![Architecture Sequence Diagram](images/ArchitectureSequenceDiagram.png)
```

Comment:
This image is not loading for me. Perhaps the ".png" is supposed to be ".PNG" instead?